src/math/Triangle.js

```diff
@@ -15,134 +18,122 @@ function Triangle( a, b, c ) {
 
 Object.assign( Triangle, {
 
-	getNormal: function () {
-
-		var v0 = new Vector3();
-
-		return function getNormal( a, b, c, target ) {
+	getNormal: function ( a, b, c, target ) {
 
-			if ( target === undefined ) {
+		if ( _v0 === undefined ) _v0 = new Vector3();
 
-				console.warn( 'THREE.Triangle: .getNormal() target is now required' );
-				target = new Vector3();
+		if ( target === undefined ) {
 
-			}
+			console.warn( 'THREE.Triangle: .getNormal() target is now required' );
+			target = new Vector3();
 
-			target.subVectors( c, b );
-			v0.subVectors( a, b );
-			target.cross( v0 );
+		}
 
-			var targetLengthSq = target.lengthSq();
-			if ( targetLengthSq > 0 ) {
+		target.subVectors( c, b );
+		_v0.subVectors( a, b );
+		target.cross( _v0 );
 
-				return target.multiplyScalar( 1 / Math.sqrt( targetLengthSq ) );
+		var targetLengthSq = target.lengthSq();
+		if ( targetLengthSq > 0 ) {
 
-			}
+			return target.multiplyScalar( 1 / Math.sqrt( targetLengthSq ) );
 
-			return target.set( 0, 0, 0 );
+		}
 
-		};
+		return target.set( 0, 0, 0 );
 
-	}(),
+	},
 
 	// static/instance method to calculate barycentric coordinates
 	// based on: http://www.blackpawn.com/texts/pointinpoly/default.html
-	getBarycoord: function () {
-
-		var v0 = new Vector3();
-		var v1 = new Vector3();
-		var v2 = new Vector3();
+	getBarycoord: function ( point, a, b, c, target ) {
 
-		return function getBarycoord( point, a, b, c, target ) {
+		if ( _v1 === undefined ) {
```

Comment:
I understand why this is not checking for `_v0`, but I think checking for `_v2` may be easier to understand.

Reply:
I mean changing the pattern to check for the last item instead of the first item (in this case the second item because of... reasons).

Reply:
Okay, makes sense. I'll adapt this pattern for future PRs 👍. `Triangle` is improved here: #17182